File: CIVITASRUBY/lib/calle.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

module Civitas
  
  require_relative 'casilla'
  require_relative 'titulo_propiedad'
  class Calle < Casilla
    attr_reader :titulo_propiedad
    def initialize(carcel,titulo)
      super(titulo.nombre,carcel)
      @titulo_propiedad = titulo
    end
    
     def recibe_jugador(actual,todos)
      if jugador_correcto(actual,todos) == true
        informe(actual,todos)
        jugador = todos[actual]
       
        if @titulo_propiedad.tiene_propietario() == false
          return jugador.puede_comprar_casilla()
        else
          return @titulo_propiedad.tramitar_alquiler(jugador)
        end
      end
    end
    
     def to_s
      result = "Casilla: \nnombre= #{@nombre} \n tipo= #{@tipo} "
      if @titulo_propiedad != nil
        result +="\n"
        result += @titulo_propiedad.to_s
      end

      result +="\n"
      
      result
     end
  end
end

File: CIVITAS/src/civitas/Calle.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;

import java.util.ArrayList;

/**
 *
 * @author Usuario
 */
public class Calle extends Casilla{
    
    private TituloPropiedad tituloPropiedad;
    
    Calle (TituloPropiedad titulo){
        super(titulo.getNombre());
        this.tituloPropiedad = titulo;
    
      
    }
    
   @Override
   public void recibeJugador(int iactual , ArrayList<Jugador> todos){
        if(this.jugadorCorrecto(iactual, todos)){
            this.informe(iactual, todos);
            Jugador jugador = todos.get(iactual);

            if(!this.tituloPropiedad.tienePropietario()){
                jugador.puedeComprarCasilla();
                
            }else{
                this.tituloPropiedad.tramitarAlquiler(jugador);
            }
        }

    } 
   
   @Override
   TituloPropiedad getTituloPropiedad() {
      
        return this.tituloPropiedad;
        
    }
   
    @Override
    public String toString() {
        String result = "\nCasilla{\n" + "nombre=" + this.getNombre();
      
        if(this.tituloPropiedad != null)
            result += this.tituloPropiedad.toString();
        
        result += "\n}";
        return result;
    }

}
